apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -13,25 +13,83 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
+/**
+ * ApolloCall is an abstraction for a request that has been prepared for execution. ApolloCall
+ * represents a single request/response pair and cannot be executed twice, though it can be cancelled.
+ */
 public interface ApolloCall<T> extends Cancelable {
 
+  /**
+   * Sends the request immediately and blocks until the response can be processed or is an error.
+   *
+   * @return The successful or failed {@link Response}
+   * @throws ApolloException       if the request could not be executed due to a cancellation, a timeout or a network
+   *                               failure
+   * @throws IllegalStateException when the call has already been executed
+   */
   @Nonnull Response<T> execute() throws ApolloException;
 
+  /**
+   * Schedules the request to be executed at some point in the future.
+   * The dispatcher defines when the request will run: usually immediately unless there are several other requests
+   * currently being executed.
+   *
+   * @param callback Callback which will handle the response or a failure exception.
+   * @throws IllegalStateException when the call has already been executed
+   */
   void enqueue(@Nullable Callback<T> callback);
 
+  /**
+   * Returns a watcher to watch the changes made by this call to the normalized cache store.
+   */
   @Nonnull ApolloWatcher<T> watcher();
 
+  /**
+   * Sets the {@link HttpCacheControl} strategy for an ApolloCall object.
+   *
+   * @param httpCacheControl the HttpCacheControl strategy to set
+   * @return The ApolloCall object with the provided HttpCacheControl strategy
+   */
   @Nonnull ApolloCall<T> httpCacheControl(@Nonnull HttpCacheControl httpCacheControl);
 
+  /**
+   * Sets the {@link CacheControl} strategy for an ApolloCall object.
+   *
+   * @param cacheControl the CacheControl strategy to set
+   * @return The ApolloCall object with the provided CacheControl strategy
+   */
   @Nonnull ApolloCall<T> cacheControl(@Nonnull CacheControl cacheControl);
 
+  /**
+   * Creates a new, identical call to this one which can be enqueued or executed even if this call
+   * has already been.
+   *
+   * @return The cloned ApolloCall object.
+   */
   @Nonnull ApolloCall<T> clone();
 
+  /**
+   * Communicates responses from a server or offline requests.
+   */
   abstract class Callback<T> {
+
+    /**
+     * Gets called when GraphQl response is received successfully.
+     *
+     * @param response the GraphQl response
```

Comment:
I feel we should use GraphQL only because the Response (which contains not just Operation but also contains data and list of errors) object represents the successful or failed response and is not solely representative of  the Operation.